### DISCUSSION TITLE
Disable client side sorting

### DISCUSSION THREAD

**Discussion by TobseF** [primefaces/community#1763] [PrimeVue Blocks]
We are using the [DataTable](https://github.com/orgs/primefaces/discussions/categories/primevue-blocks) component to display paginated and sorted data.
To sort columns we make a column sortable and update the data on the sort event  `@sort="onColumnSort"`.
But how can we disable the client side sorting? The _onColumnSort_ triggers an Update and fetches the new data, already sorted by the SQL query.
Is there any way to use the sort indicator only as indicator and not with it's client side sorting functionality - what did I missed? 

The problem is, that the client sorts the data in other ways than the database query. For example, the database can use the ordinal of an enum for the sort. The client can only use the text value. In case of an error state label, this is not intended. In addition loading the table data triggers the sorting after the data is loaded and displayed. So the user just sees the data, sorted by the db, for a half second, befor it'S get sorted by the DataTable.

![2024-04-25_12h50_06](https://github.com/primefaces/community/assets/1190109/e8b678cd-866f-4bce-9d3a-e5b176362a02)

**Comment by TobseF** [comment]
I got an answer from SlavCo that it's not possibel at the moment.
> I don't think you can opt in for sorting indicators in the column headers but opt out from the sorting behaviour... at least not that I can immidiatley see.
> I'd suggest to use  header slot of the Column component and pass your own sorting indicators which you can be in full control of
> https://primevue.org/datatable/#api.column.slots.header

 I'm a little bit confused, because for me it seems a default usecase for many projects: Load data based on query parameters and display it paged. Then client side sorting is an additional opt in feature, which can be added for some edge cases, were it's possible to deliver all rows at once. Implementing it should be very easy... just skip the sorting based on some additional flag. It seems I need to raise an feature request.

**Reply by Fisujon** [reply]
Did you find any solution?

**Comment by vldfokin** [comment]
A bit late to the party, but for anyone else looking for a solution: I believe you can achieve this behavior by using the [lazy](https://primevue.org/datatable/#api.datatable.props.lazy) prop.

**Reply by mashal02** [reply]
yep! :lazy=true worked for me, I had the same problem which @TobseF mentioned.
